CMake/FindSDL2_net.cmake

```diff
@@ -0,0 +1,197 @@
+# - Find SDL2_net
+# Find the SDL2 headers and libraries
+#
+#  SDL2::SDL2_net - Imported target
+#
+#  SDL2_net_FOUND - True if SDL2_net was found.
+#  SDL2_net_DYNAMIC - If we found a DLL version of SDL2_net
+#
+# Modified for SDL2_net of FindSDL2.cmake
+# Original Author:
+# 2015 Ryan Pavlik <ryan.pavlik@gmail.com> <abiryan@ryand.net>
+#
+# Copyright Sensics, Inc. 2015.
+# Distributed under the Boost Software License, Version 1.0.
+# (See accompanying file LICENSE_1_0.txt or copy at
+# http://www.boost.org/LICENSE_1_0.txt)
+
+# Set up architectures (for windows) and prefixes (for mingw builds)
+if(WIN32)
+	if(MINGW)
+		include(MinGWSearchPathExtras OPTIONAL)
+		if(MINGWSEARCH_TARGET_TRIPLE)
+			set(SDL2_net_PREFIX ${MINGWSEARCH_TARGET_TRIPLE})
+		endif()
+	endif()
+	if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+		set(SDL2_net_LIB_PATH_SUFFIX lib/x64)
+		if(NOT MSVC AND NOT SDL2_net_PREFIX)
+			set(SDL2_net_PREFIX x86_64-w64-mingw32)
+		endif()
+	else()
+		set(SDL2_net_LIB_PATH_SUFFIX lib/x86)
+		if(NOT MSVC AND NOT SDL2_net_PREFIX)
+			set(SDL2_net_PREFIX i686-w64-mingw32)
+		endif()
+	endif()
+endif()
+
+if(SDL2_net_PREFIX)
+	set(SDL2_net_ORIGPREFIXPATH ${CMAKE_PREFIX_PATH})
+	if(SDL2_net_ROOT_DIR)
+		list(APPEND CMAKE_PREFIX_PATH "${SDL2_net_ROOT_DIR}")
+	endif()
+	if(CMAKE_PREFIX_PATH)
+		foreach(_prefix ${CMAKE_PREFIX_PATH})
+			list(APPEND CMAKE_PREFIX_PATH "${_prefix}/${SDL2_net_PREFIX}")
+		endforeach()
+	endif()
+	if(MINGWSEARCH_PREFIXES)
+		list(APPEND CMAKE_PREFIX_PATH ${MINGWSEARCH_PREFIXES})
+	endif()
+endif()
+
+# Invoke pkgconfig for hints
```

Comment:
Looks like this was based on the SDL_ttf version that we have but that version is somewhat broken.
If SDL_net is found via pkg-config, it should not be used for hints but directly

```cmake
pkg_check_modules(SDL2_net QUIET SDL2_net IMPORTED_TARGET)
```

This will create a `PkgConfig::SDL2_net` imported target with the correct link flags and include directories.